MariaDB/server
# MDEV-28376: If Perl DBD::MariaDB is no available test for Perl DBD::mysl in mariadb-report

- [x] *The Jira issue number for this PR is: MDEV-28376*

## Description
Commit introduces automatic detection which DBI driver is available (and bail out if non of them is).

Current Detection prefers **Perl DBD:MariaDB** driver.

This is mainly for older Linux distros or Windows which does not have Perl DBD:MariaDB packaged or does not want to use Perl cpan command.


## How can this PR be tested?
Install for example Ubuntu Jammy (22.04). Install perl DBD:mysql: `apt install libclass-dbi-mysql-perl`
run `debian/additions/mariadb-report` (with MariaDB 10.5 running in background) and it should  connect happily to server without errors. If you do it with  without this commit it should die on:
```
failed: Can't locate DBD/MariaDB.pm in @INC (you may need to install the DBD::MariaDB module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.34.0 /usr/local/share/perl/5.34.0 /usr/lib/x86_64-linux-gnu/perl5/5.34 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.34 /usr/share/perl/5.34 /usr/local/lib/site_perl) at (eval 9) line 3.
```

## Basing the PR against the correct MariaDB version

## Backward compatibility
Brings back more backward compatibility as Perl DBD:mysql is supported.
